third_party/llvm-openmp/BUILD

```
@@ -0,0 +1,124 @@
+# Build file for OpenMP library that is part of llvm
+
+exports_files(["LICENSE.txt"])
+
+load(
+    "@org_tensorflow//third_party/llvm:llvm.bzl",
+    "cmake_var_string",
+    "dict_add",
+    "expand_cmake_vars",
+)
+load(
+    "@org_tensorflow//third_party:common.bzl",
+    "template_rule",
+)
+
+genrule(
+    name = "il8n_id",
+    srcs = [
+        "runtime/tools/message-converter.pl",
+        "runtime/src/i18n/en_US.txt",
+    ],
+    outs = ["include/kmp_i18n_id.inc"],
+    cmd = "$(location runtime/tools/message-converter.pl) --os=lin --prefix=kmp_i18n --enum=$@  $(location runtime/src/i18n/en_US.txt)",
+)
+
+genrule(
+    name = "kmp_i18n_default",
+    srcs = [
+        "runtime/tools/message-converter.pl",
+        "runtime/src/i18n/en_US.txt",
+    ],
+    outs = ["include/kmp_i18n_default.inc"],
+    cmd = "perl $(location runtime/tools/message-converter.pl) --os=lin --prefix=kmp_i18n --default=$@ $(location runtime/src/i18n/en_US.txt)",
+)
+
+# Bazel doesn't accept .txt as an input, rename the ldscript to .inc to workaround.
+genrule(
+    name = "ldscript",
+    srcs = ["runtime/src/exports_so.txt"],
+    outs = ["exports_so.inc"],
+    cmd = "cp $(location runtime/src/exports_so.txt) $@",
+)
+
+# Cmake vars to replace.
+omp_vars = {
+    "LIBOMP_USE_VERSION_SYMBOLS": 1,
+    "LIBOMP_HAVE_WEAK_ATTRIBUTE": 1,
+    "LIBOMP_USE_ADAPTIVE_LOCKS": 1,
+    "LIBOMP_ENABLE_ASSERTIONS": 1,
+    "LIBOMP_ENABLE_SHARED": 1,
+    "LIBOMP_LEGAL_ARCH": "Intel(R) 64",
+    "LIBOMP_LIB_FILE": "libiomp5.so",
+    "LIBOMP_VERSION_MAJOR": 5,
+    "LIBOMP_VERSION_MINOR": 0,
+}
+
+omp_all_cmake_vars = cmake_var_string(dict_add(omp_vars))
+
+expand_cmake_vars(
+    name = "config_kmp",
+    src = "runtime/src/kmp_config.h.cmake",
+    cmake_vars = omp_all_cmake_vars,
+    dst = "include/kmp_config.h",
+)
+
+expand_cmake_vars(
+    name = "config_omp",
+    src = "runtime/src/include/omp.h.var",
+    cmake_vars = omp_all_cmake_vars,
+    dst = "include/omp.h",
+)
+
+cc_binary(
+    name = "libiomp5.so",
```

Nit: The target name doesn't need to be the same as the output name.
```suggestion
    name = "libiomp5",
```

---

@penpornk bazel requires the .so extension for linux builds.

---

I just realized that you are using `cc_binary` target instead of `cc_library`. Files are not not supposed to be executed on their own should be a `cc_library`. Could you please try with [cc_library](https://docs.bazel.build/versions/master/be/c-cpp.html#cc_library)?

---

Chatted offline. It seems this is nontrivial to solve. In the interest of time, we'll proceed with `libiomp5` being `cc_binary` for now.

Could you please add a TODO (over the `cc_binary` line to convert to `cc_library` later? It'd be great if you could also paste the error there:
```
ERROR: //tensorflow/BUILD:689:1: Linking of rule '//tensorflow:libtensorflow_framework.so.2.4.0' failed (Exit 1)
/usr/bin/ld.gold: error: symbol GOMP_parallel_loop_nonmonotonic_guided has undefined version VERSION
/usr/bin/ld.gold: error: symbol GOMP_parallel_start has undefined version GOMP_1.0
/usr/bin/ld.gold: error: symbol GOMP_cancellation_point has undefined version GOMP_4.0
/usr/bin/ld.gold: error: symbol omp_set_num_threads has undefined version OMP_1.0
/usr/bin/ld.gold: error: symbol GOMP_parallel_loop_nonmonotonic_dynamic has undefined version GOMP_4.5
/usr/bin/ld.gold: error: symbol omp_get_partition_place_nums has undefined version OMP_4.5
/usr/bin/ld.gold: error: symbol omp_unset_lock has undefined version OMP_3.0
/usr/bin/ld.gold: error: symbol GOMP_loop_ull_static_next has undefined version GOMP_2.0
/usr/bin/ld.gold: error: symbol omp_is_initial_device_ has undefined version OMP_4.0
/usr/bin/ld.gold: error: symbol omp_get_wtime_ has undefined version OMP_2.0
/usr/bin/ld.gold: error: symbol omp_in_final_ has undefined version OMP_3.1
/usr/bin/ld.gold: error: symbol GOMP_taskyield has undefined version GOMP_3.0
```